Fix WaitForLock() and config.Set() lock (#1370)
ActiveState/cli: changelog.md

  will from now on be made available through the system tray application and you
  will have the choice to install it now or later.

### Fixed

> - Fixed logic checking for deprecations.

Reviewer: Remember to phrase this with end users in mind. What does this fix mean practically for end-users?

  will from now on be made available through the system tray application and you
  will have the choice to install it now or later.

### Fixed

> - Fixed a rare case where deprecation information was not downloaded, when
>   requested by several State Tool instances in parallel. This also prevents the
>   configuration file from getting corrupted when two processes access it
>   simulataneously.  ([PR #1370] (https://github.com/ActiveState/cli/pull/1370))

Reviewer: ```suggestion
- Fixed a rare case where the configuration file can get corrupted when two processes access it
  simultaneously.  ([PR #1370] (https://github.com/ActiveState/cli/pull/1370))
```
